### ISSUE TITLE
Yarn support

### ISSUE THREAD

**Issue by Apidcloud** [aredridel/node-bin-gen#44]
Good evening!

I've noticed `yarn add node@8.9.4` doesn't really add the binaries to `node_modules/.bin`, whereas npm does. Would it be possible to support yarn as well?

Thanks!

**Comment by aredridel** [comment]
Possibly! how does yarn install binaries?

**Comment by Apidcloud** [comment]
You mean the command to do so? `yarn add package_x`

**Comment by aredridel** [comment]
Mmm. I don't use yarn at all, so I'm not sure with how it differs from how npm installs packages. Where does it put binaries if not in node_modules/.bin?

(I can do the digging, but any help and context you can give will speed it up!)

**Comment by Apidcloud** [comment]
Hmm, it should be inside .bin as well. My first guess is that it's not running the preinstall script at all (?). Not sure if the current version uses the same approach?

**Comment by aredridel** [comment]
Oh interesting! Does yarn generally skip scripts like that?

**Comment by Apidcloud** [comment]
No, not really. Just did a quick test and it calls a "pre" script just fine.

**Comment by aredridel** [comment]
Interesting. I'll have to dig into why, unless someone else gets to it first!

**Comment by Apidcloud** [comment]
Thanks a lot for taking the time to! 😄

**Comment by vkrol** [comment]
The cause of the problem is https://github.com/yarnpkg/yarn/issues/3421. The workaround is https://github.com/yarnpkg/yarn/issues/3421#issuecomment-390230608.

**Comment by nicolo-ribaudo** [comment]
Note that this package doesn't work with PnP, which is the "algorithm" used by Yarn 2 and which is opt-in in yarn 1: https://github.com/yarnpkg/berry/issues/637#issuecomment-566518710
